File: data/evaluation/geeks_for_geeks_successful_test_scripts/python/SWAP_BITS_IN_A_GIVEN_NUMBER.py
# Copyright (c) 2019-present, Facebook, Inc.
# All rights reserved.
#
# This source code is licensed under the license found in the
# LICENSE file in the root directory of this source tree.
#
def f_gold ( x , p1 , p2 , n ) :
    set1 = ( x >> p1 ) & ( ( 1 << n ) - 1 )
    set2 = ( x >> p2 ) & ( ( 1 << n ) - 1 )
    xor = ( set1 ^ set2 )
    xor = ( xor << p1 ) | ( xor << p2 )
    result = x ^ xor
    return result


#TOFILL

if __name__ == '__main__':
    param = [
    (95,88,97,92,),
    (16,26,59,42,),
    (55,56,40,41,),
    (75,35,79,30,),
    (90,12,59,34,),
    (58,65,25,19,),
    (69,64,17,94,),
    (5,1,59,38,),
    (36,33,97,44,),
    (62,69,66,9,)
        ]
    n_success = 0
    for i, parameters_set in enumerate(param):
        if f_filled(*parameters_set) == f_gold(*parameters_set):
            n_success+=1
    print("#Results: %i, %i" % (n_success, len(param)))

File: data/evaluation/geeks_for_geeks_successful_test_scripts/java/SWAP_BITS_IN_A_GIVEN_NUMBER.java
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

import java.util. *;
import java.util.stream.*;
import java.lang.*;
import javafx.util.Pair;
public class SWAP_BITS_IN_A_GIVEN_NUMBER{
static int f_gold ( int x , int p1 , int p2 , int n ) {
  int set1 = ( x >> p1 ) & ( ( 1 << n ) - 1 ) ;
  int set2 = ( x >> p2 ) & ( ( 1 << n ) - 1 ) ;
  int xor = ( set1 ^ set2 ) ;
  xor = ( xor << p1 ) | ( xor << p2 ) ;
  int result = x ^ xor ;
  return result ;
}


//TOFILL

public static void main(String args[]) {
    int n_success = 0;
    List<Integer> param0 = new ArrayList<>();
    param0.add(95);
    param0.add(16);
    param0.add(55);
    param0.add(75);
    param0.add(90);
    param0.add(58);
    param0.add(69);
    param0.add(5);
    param0.add(36);
    param0.add(62);
    List<Integer> param1 = new ArrayList<>();
    param1.add(88);
    param1.add(26);
    param1.add(56);
    param1.add(35);
    param1.add(12);
    param1.add(65);
    param1.add(64);
    param1.add(1);
    param1.add(33);
    param1.add(69);
    List<Integer> param2 = new ArrayList<>();
    param2.add(97);
    param2.add(59);
    param2.add(40);
    param2.add(79);
    param2.add(59);
    param2.add(25);
    param2.add(17);
    param2.add(59);
    param2.add(97);
    param2.add(66);
    List<Integer> param3 = new ArrayList<>();
    param3.add(92);
    param3.add(42);
    param3.add(41);
    param3.add(30);
    param3.add(34);
    param3.add(19);
    param3.add(94);
    param3.add(38);
    param3.add(44);
    param3.add(9);
    for(int i = 0; i < param0.size(); ++i)
    {
        if(f_filled(param0.get(i),param1.get(i),param2.get(i),param3.get(i)) == f_gold(param0.get(i),param1.get(i),param2.get(i),param3.get(i)))
        {
            n_success+=1;
        }
    }
    System.out.println("#Results:" + n_success + ", " + param0.size());
}
}